src/utilities/include/metaphysicl/dynamic_std_array_wrapper.h

```diff
@@ -128,13 +128,13 @@ class DynamicStdArrayWrapper
   reverse_iterator rbegin()
   {
     metaphysicl_assert(_dynamic_n <= N);
-    return _data.rbegin() + (N - _dynamic_n);
+    return _data.rend() + _dynamic_n;
```

Comment:
Can this be right?  Isn't rend() a reverse iterator?  You're not going to get to rbegin() from it by adding a positive number, are you?

I think we need some unit test coverage of these either way.

Reply:
you use a reverse iterator like this:

```
for (std::vector<int>::reverse_iterator it = v.rbegin(); it != v.rend(); ++it)
```

i.e. the iterator is still advanced with `++`

Reply:
Well, what can I say. _Of course_ you were right. It's `- _dynamic_n`! I added a unit test.